rust/main/lander/src/adapter/chains/ethereum/nonce/updater.rs

```diff
@@ -0,0 +1,75 @@
+use std::future::Future;
+use std::ops::Add;
+use std::sync::Arc;
+use std::time::Duration;
+
+use ethers_core::types::Address;
+use tokio::sync::Mutex;
+use tokio::time::{sleep, Instant};
+use tokio_metrics::TaskMonitor;
+use tracing::{error, info, info_span, Instrument};
+
+use hyperlane_core::U256;
+use hyperlane_ethereum::{EthereumReorgPeriod, EvmProviderForLander};
+
+use super::state::NonceManagerState;
+
+pub struct NonceUpdater {
+    address: Address,
+    reorg_period: EthereumReorgPeriod,
+    block_time: Duration,
+    provider: Arc<dyn EvmProviderForLander>,
+    state: Arc<NonceManagerState>,
+    updated: Arc<Mutex<Instant>>,
+}
+
+impl NonceUpdater {
```

Comment:
given this background task wasn't in the original design and seems more like on optimization, I'd be in favor of lazily reading the nonce from onchain every N submissions, like we currently do in our ethers NonceManager